bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -322,14 +322,21 @@ def cmake_enable_option(spec_requiremement):
             options.append("-DCORENRN_NMODL_FLAGS=%s" % nmodl_options)
 
             if spec.satisfies("+gpu"):
-                gcc = which("gcc")
                 nvcc = which("nvcc")
-                options.extend(
-                    ["-DCMAKE_CUDA_COMPILER=%s" % nvcc, "-DCMAKE_CUDA_HOST_COMPILER=%s" % gcc]
-                )
+                # Instead of assuming that the gcc in $PATH is the right host compiler, take the
+                # compiler used to build the cuda package as the CUDA host compiler.
+                host_compiler_spec = self.spec["cuda"].compiler
+                # Surely this isn't the best way
+                host_compiler_candidates = [
+                    c for c in spack.compilers.all_compilers() if c.spec == host_compiler_spec
+                ]
+                assert len(host_compiler_candidates) == 1
+                host_compiler = host_compiler_candidates[0]
```

Comment:
@matz-e this is..ugly, but I didn't see another way.
This is the alternative to hacking the compiler config in `bin/configure-compilers` to add the relevant GCC module to the list that needs to be loaded for NVHPC.

Reply:
I agree this is ugly - having this logic in spack recipe 😒. Shouldn't this be enforced by the spec constraint itself?

Reply:
I didn't see a simpler way.

When we build `neuron%nvhpc` then we need a CUDA compiler (`nvcc`) for the few `.cu` files we have lying around. We get this via the dependency on `cuda`. For better or for worse, installing `cuda%gcc@12` does not hardcode/configure `nvcc` to use `gcc@12` as its host compiler -- unlike `nvc++` -- so when we actually use `nvcc` (i.e. when building `neuron`) we have to tell it what host compiler to use.

When we're building `neuron%nvhpc`, the only C++ compiler that Spack offers us is `nvc++`, and we generally don't use that as the `nvcc` host compiler, so <some magic> is needed to find the path to `g++` 12.